### PR TITLE
Fixed Multi ALM enhancement Issues

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

File: Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs [hunk 1]
```diff
@@ -221,7 +222,7 @@ private bool GetProjectsDetails()
             {
                 isProjectMappingCorrect = false;
             }
-            SetControls();
+            //SetControls();
```
Comment (ambarkhan) [inline Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs:225]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 2]
```diff
@@ -74,7 +74,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMPassword) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectKey) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:80]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688&bid=19696944)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 3]
```diff
@@ -74,7 +74,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:78]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 4]
```diff
@@ -74,7 +74,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:77]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 5]
```diff
@@ -74,7 +74,18 @@ public override bool ConnectALMServer()
...
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMPassword) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:79]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 6]
```diff
@@ -74,7 +74,18 @@ public override bool ConnectALMServer()
...
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectKey) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectName) == false)
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:81]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688&bid=19696944)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 7]
```diff
@@ -74,7 +72,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMPassword) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectKey) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:78]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 8]
```diff
@@ -74,7 +72,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:76]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 9]
```diff
@@ -74,7 +72,18 @@ public override bool ConnectALMServer()
         public override bool ConnectALMProject()
         {
             ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
-            return true;
+            if (string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMServerURL) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:75]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 10]
```diff
@@ -74,7 +72,18 @@ public override bool ConnectALMServer()
...
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMUserName) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMPassword) == false &&
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:77]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs [hunk 11]
```diff
@@ -74,7 +72,18 @@ public override bool ConnectALMServer()
...
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectKey) == false &&
+                string.IsNullOrEmpty(ALMCore.DefaultAlmConfig.ALMProjectName) == false)
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:79]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs:225]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688&bid=19696944)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 8 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:76]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 11 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:79]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 9 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:75]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 10 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:77]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs [hunk 12]
```diff
@@ -569,11 +575,15 @@ private void ALMRadioButton_Checked_Changed(object sender, RoutedEventArgs e)
                 //Bind again as we changed the AlmConfig object
                 Bind();
 
-                StyleRadioButtons();
-
+                //clear items
+                //DomainComboBox.Items.Clear();
```
Comment (ambarkhan) [inline Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs:579]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 7 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:78]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

**Review by JinendraGhodke — APPROVED** [review]

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688&bid=19696944)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 9 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:75]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 7 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:78]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 10 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:77]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 8 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:76]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/GingerCore/ALM/QtestCore.cs (hunk 11 above)
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/QtestCore.cs:79]: Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

File: Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs (hunk 12 above)
Comment (ambarkhan) [inline Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs:579]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6008688)

**Review by JinendraGhodke — APPROVED** [review]

**Pull request merged**